cli/src/commands/fix.rs

```diff
@@ -328,3 +382,111 @@ fn run_tool(
         Err(())
     }
 }
+
+/// Represents an entry in the `fix.tools` config table.
+struct ToolConfig {
+    /// The command that will be run to fix a matching file.
+    command: CommandNameAndArgs,
+    /// The matcher that determines if this tool matches a file.
+    matcher: Box<dyn Matcher>,
+    // TODO: Store the `name` field here and print it with the command's stderr, to clearly
+    // associate any errors/warnings with the tool and its configuration entry.
+}
+
+/// Represents the `fix.tools` config table.
+struct ToolsConfig {
+    /// Some tools, stored in the order they will be executed if more than one
+    /// of them matches the same file.
+    tools: Vec<ToolConfig>,
+}
+
+/// Simplifies deserialization of the config values while building a ToolConfig.
+#[derive(Clone, Debug, Default, Eq, PartialEq, serde::Deserialize)]
+#[serde(default)]
+struct RawToolConfig {
+    command: Vec<String>,
+    patterns: Vec<String>,
+}
```

Comment:
nit: `CommandNameAndArgs` can be embedded (if it's okay to support string and structured forms)
```rust
#[derive(Clone, Debug, Eq, PartialEq, serde::Deserialize)]
#[serde(rename_all = "kebab-case")]
struct RawToolConfig {
    command: CommandNameAndArgs,
    patterns: Vec<String>,
}
```

Reply:
I don't think this works, because `CommandNameAndArgs` doesn't implement `Default`. Am I missing something?

Reply:
Do we need `Default` at all? Both `command` and `patterns` are required.

Reply:
I see what you mean now. Is it true that `Matcher` would still need some work to support `Deserialize` like this?

Reply:
`Matcher` won't fit to `Deserialize` because it needs some environment parameters to build, I suppose. (Maybe could be build with `Fs { cwd: "", base: "" }`, but it seems bad idea to encode a specific environment into the type.)